controller/k8s/api.go

```diff
@@ -256,9 +257,10 @@ func (api *API) Sync(stopCh <-chan struct{}) {
 
 	log.Infof("waiting for caches to sync")
 	if !cache.WaitForCacheSync(ctx.Done(), api.syncChecks...) {
-		log.Fatal("failed to sync caches")
+		return errors.New("failed to sync caches")
```

Comment:
This will require changes on all the controllers calling `Sync()` that are not catching this error and will now continue running despite the API not being ready.